sunpy/data/_sample.py

```diff
@@ -116,9 +116,9 @@ def get_sample_file(filename, url_list, show_progress=True, overwrite=False,
     else:
         uncompressed_filename = filename
     # check if the (uncompressed) file exists
-    if not overwrite and os.path.isfile(os.path.join(sampledata_dir,
-                                                     uncompressed_filename)):
-        return os.path.join(sampledata_dir, uncompressed_filename)
+    if not overwrite and Path(sampledata_dir).joinpath(
+                        uncompressed_filename).is_file():
```

Comment:
This and below would be simpler if `get_and_create_sample_dir` return a `Path`

Reply:
could you give me an idea where all you want me to return Path objects

Reply:
He mentioned it, `get_and_create_sample_dir`.